zkvm/src/ops.rs

```diff
@@ -241,7 +242,13 @@ impl Instruction {
             }
             Instruction::Const => write(Opcode::Const),
             Instruction::Var => write(Opcode::Var),
-            Instruction::Alloc => write(Opcode::Alloc),
+            Instruction::Alloc(osw) => {
+                write(Opcode::Alloc);
+                match osw {
```

Comment:
The assignment is the private data held by the prover, it's not part of the bytecode. Only the Opcode::Alloc needs to be written to the bytecode string.